packages/client/src/App.tsx

```diff
@@ -1,21 +1,27 @@
-// import { useEffect } from 'react';
-// import React from 'react';
-import { Route, Routes } from 'react-router-dom'
+import { Route, Routes, useLocation } from 'react-router-dom'
 import { MainLayout } from './components/MainLayout'
 import { Main } from './pages/Main'
+import { Login } from './pages/Login'
+import { Registration } from './pages/Registration'
+import { Profile } from './pages/Profile'
+import { Leaderboard } from './pages/Leaderboard'
+import { NotFound } from './pages/NotFound'
+import { Forum } from './pages/Forum'
 
 import './App.css'
 
 function App() {
+  const pathname = useLocation().pathname
   return (
-    <MainLayout>
+    <MainLayout background={`${pathname}`}>
```

Comment:
useLocation в кач-ве pathname возвращает строку, поэтому можно сразу отправить ее в проп
```suggestion
    <MainLayout background={pathname}>
```